ibm/data_source_ibm_is_vpn_gateway_connections.go

```diff
@@ -0,0 +1,234 @@
+						},
+						isVPNGatewayConnectionIPSECPolicy: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "IP security policy for vpn gateway connection",
+						},
+						isVPNGatewayConnectionMode: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The mode of the VPN gateway",
+						},
+						isVPNGatewayConnectionName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "VPN Gateway connection name",
+						},
+						isVPNGatewayConnectionPeerAddress: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "VPN gateway connection peer address",
+						},
+						isVPNGatewayConnectionResourcetype: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The resource type",
+						},
+						isVPNGatewayConnectionStatus: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "VPN gateway connection status",
+						},
+
+						isVPNGatewayConnectionTunnels: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Optional:    true,
+							MinItems:    0,
+							Description: "The VPN tunnel configuration for this VPN gateway connection (in static route mode)",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"address": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The IP address of the VPN gateway member in which the tunnel resides",
+									},
+
+									"status": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The status of the VPN Tunnel",
+									},
+								},
+							},
+						},
+						isVPNGatewayConnectionLocalCIDRS: {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Set:         schema.HashString,
+							Description: "VPN gateway connection local CIDRs",
+						},
+
+						isVPNGatewayConnectionPeerCIDRS: {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Set:         schema.HashString,
+							Description: "VPN gateway connection peer CIDRs",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMVPNGatewayConnectionsRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	vpngatewayID := d.Get(isVPNGatewayID).(string)
+	listvpnGWConnectionOptions := sess.NewListVPNGatewayConnectionsOptions(vpngatewayID)
+
+	if gwstatus, ok := d.GetOk(isVPNGatewayConnectionsStatus); ok {
+		listvpnGWConnectionOptions.SetStatus(gwstatus.(string))
+	}
+	availableVPNGatewayConnections, detail, err := sess.ListVPNGatewayConnections(listvpnGWConnectionOptions)
+	if err != nil {
```

Comment:
Can you check ListVPNGatewayConnections also you need to add looping and storing all results